news/20200723142705.doc

```diff
@@ -0,0 +1 @@
+Add ADR for removing mbed_config.h
```

Comment:
ADRs probably don't need NEWS items, as they are developer-facing rather than user-facing. NEWS is for users; developers can use `git log`.

Reply:
We should tweak the CI so it doesn't try to fail the PR when there's no news fragment.